autoACDC/autoACDC.py

```diff
@@ -0,0 +1,501 @@
+                    ancestor_wf = None
+                    break
+                ancestor_wf = workflowInfo(self.url, ancestor_wf.request['OriginalRequestName'])
+        else:
+            raise Exception("'OriginalRequestName' not in schema.")
+
+        return original_wf, ancestor_wf
+
+    def getAssignParameters(self):
+        """
+        Defines a dictionary of paremeters for assign based on the
+        original workflow, the ACDC created, and the options that
+        are passed to the class.
+        Returns: a dictionary of parameters for assign.
+        """
+
+        self.setACDCWfInfo()
+
+        original_wf, ancestor_wf = self.getPreviousWfs()
+
+        # check to see if the workflow is a task chain or an ACDC of a taskchain
+        taskchain = (self.schema["RequestType"] == "TaskChain") or (ancestor_wf and ancestor_wf.request["RequestType"] == "TaskChain")
+
+        # these are automatically determined from ancestor workflow
+        era = ancestor_wf.acquisitionEra()
+        procstring = ancestor_wf.processingString()
+        if (not era or not procstring) or (taskchain and (type(era)!=dict or type(procstring)!=dict)):
+            raise Exception("We do not have a valid AcquisitionEra and ProcessingString")
+        procversion = ancestor_wf.request['ProcessingVersion']
+        
+        # some default options in case the arguments aren't passed
+
+        # Must use --lfn option, otherwise workflow won't be assigned
+        if self.options['lfn']:
+            lfn = self.options['lfn']
+        elif "MergedLFNBase" in self.schema:
+            lfn = self.schema['MergedLFNBase']
+        elif ancestor_wf and "MergedLFNBase" in ancestor_wf.request:
+            lfn = ancestor_wf.request['MergedLFNBase']
+        else:
+            raise Exception("Can't assign the workflow! Please include workflow lfn using --lfn option.")
+        
+        # activity production by default for taskchains, reprocessing for default by workflows
+        if self.options['activity']:
+            activity = self.options['activity']
+        elif taskchain:
+            activity = 'production'
+        else:
+            activity = 'reprocessing'   
+
+        # inherit or overwrite
+        if self.options['secondary_xrootd']:
+            secondary_xrootd =  self.options['secondary_xrootd']
+        elif ancestor_wf and "TrustPUSitelists" in ancestor_wf.request:
+            secondary_xrootd = ancestor_wf.request['TrustPUSitelists']
+        else:
+            secondary_xrootd = False
+
+        params = {
+            "SiteWhitelist": self.getSites(),
+            "MergedLFNBase": lfn,
+            "Dashboard": activity,
+            "ProcessingVersion": procversion,
+            "execute": True,
+            "AcquisitionEra": era,
+            "ProcessingString": procstring,
+            "TrustSitelists": self.options['xrootd'],
+            "TrustPUSitelists": secondary_xrootd
+        }
+
+        if self.options['replica']:
```

Comment:
@amaltaro  do these parameters have any use nowadays?
1. `NonCustodialSites`
2. `NonCustodialSubType`
3. `AutoApproveSubscriptionSites`

I feel like we should just drop those. What do you think?